### PR TITLE
Update browsers provided by dev-infra and introduce fallback mirror URLs to reduce flakiness

### PR DESCRIPTION
**Pull request by devversion**
See individual commits.

Port of https://github.com/angular/angular/pull/42932

### CODE REVIEW

File: bazel/browsers/chromium/find-stable-revision-for-all-platforms.ts [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+  // Note: We cannot extract the Chromium version and commit automatically because
+  // this requires an actual browser resolving a manual `window.open` redirect.
+  console.info('Release Info:', await getReleaseInfoUrlForRevision(availableRevision));
+  console.info('Click on the link above to determine the Chromium version number.');
+  console.info();
+
+  for (const platformName of Object.keys(BrowserPlatform)) {
+    const platform = BrowserPlatform[platformName as keyof typeof BrowserPlatform];
+
+    console.info(
```
[the remark this answers is not in the document]
Comment (devversion) [reply bazel/browsers/chromium/find-stable-revision-for-all-platforms.ts:98]: yeah, definitely true. It's a rarely used script so it should be okay not having great visuals here

File: bazel/browsers/chromium/find-stable-revision-for-all-platforms.ts [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+      `${platformName}: `.padEnd(10),
+      getRevisionArchiveUrl(availableRevision, platform, 'browser'),
+    );
+    console.info(
+      ' '.repeat(15),
+      await getSha256ChecksumForPlatform(availableRevision, platform, 'browser'),
+    );
+    console.info(
+      ' '.repeat(10),
+      await getRevisionArchiveUrl(availableRevision, platform, 'driver'),
```
[the remark this answers is not in the document]
Comment (devversion) [reply bazel/browsers/chromium/find-stable-revision-for-all-platforms.ts:108]: yeah, good catch.

**Comment by devversion** [comment]
@josephperrott Addressed your feedback. thanks!

**Review by josephperrott — APPROVED** [review]
LGTM

**Pull request closed without merging**